ssl/ssl_lib.c

```diff
@@ -3181,6 +3181,15 @@ SSL_CTX *SSL_CTX_new_ex(OSSL_LIB_CTX *libctx, const char *propq,
     if (ret == NULL)
         goto err;
 
+    /* Init the reference counting before any call to SSL_CTX_free */
+    ret->references = 1;
+    ret->lock = CRYPTO_THREAD_lock_new();
+    if (ret->lock == NULL) {
+        ERR_raise(ERR_LIB_SSL, ERR_R_MALLOC_FAILURE);
+        OPENSSL_free(ret);
+        return NULL;
```

Comment:
one could argue that this should be `goto err` in line with the earlier and later code flows.

Reply:
.. so it will add more code for only one case , in one call point of the code.
A lot of other new functions are just calling  OPENSSL_free(), when the object is not enough constructed to be free by his type-dedicated method.